sdk/servicebus/azure-servicebus/azure/servicebus/_common/auto_lock_renewer.py

```diff
@@ -91,23 +96,54 @@ def __init__(
         :type max_workers: Optional[int]
         """
         self._executor = executor or ThreadPoolExecutor(max_workers=max_workers)
+        # None indicates it's unknown whether the provided executor has max worker > 1
+        self._is_max_workers_greater_than_one = None if executor else (max_workers is None or max_workers > 1)
         self._shutdown = threading.Event()
-        self._sleep_time = 1
+        self._sleep_time = 0.5
         self._renew_period = 10
+        self._running = threading.Event()  # indicate whether the main worker thread is running
+        self._last_activity_timestamp = None  # the last timestamp when the main worker is active dealing with tasks
+        self._idle_timeout = 10  # the idle time that main worker thread should exist if there's no activity
```

Comment:
as oob discussion: not exposing it now to avoid confusion to customers, could expose if there're customer requests.

also update the timeout to 5 seconds.